src/full/Agda/Interaction/Highlighting/Common.hs

```diff
@@ -3,14 +3,38 @@
 module Agda.Interaction.Highlighting.Common
   ( toAtoms
   , chooseHighlightingMethod
+  , highlightOnlyCode
+  , highlightedFileExt
   ) where
 
+import Agda.Interaction.Options
 import Agda.Interaction.Highlighting.Precise
 import Agda.Syntax.Common
 import Agda.TypeChecking.Monad (HighlightingMethod(..))
 import Data.Maybe (maybeToList)
 import Data.Char (toLower)
 
+-- | Determine how to highlight the file
+highlightOnlyCode :: HtmlHighlight -> FileType -> Bool
+highlightOnlyCode HighlightAll  _ = False
+highlightOnlyCode HighlightCode _ = True
+highlightOnlyCode HighlightAuto AgdaFileType = False
+highlightOnlyCode HighlightAuto MdFileType   = True
+highlightOnlyCode HighlightAuto RstFileType  = True
+highlightOnlyCode HighlightAuto TexFileType  = False
+
+-- | Determine the generated file extension
+highlightedFileExt :: HtmlHighlight -> FileType -> String
+highlightedFileExt HighlightAll  _            = "html"
+highlightedFileExt HighlightCode AgdaFileType = "html"
+highlightedFileExt HighlightCode MdFileType   = "md"
+highlightedFileExt HighlightCode RstFileType  = "rst"
+highlightedFileExt HighlightCode TexFileType  = "tex"
+highlightedFileExt HighlightAuto AgdaFileType = "html"
+highlightedFileExt HighlightAuto MdFileType   = "md"
+highlightedFileExt HighlightAuto RstFileType  = "rst"
+highlightedFileExt HighlightAuto TexFileType  = "html"
```

Comment:
It seems that the following always holds (and should hold):

```haskell
highlightedFileExt HighlightAuto x == if highlightOnlyCode x 
                                     then highlightedFileExt HighlightCode x 
                                     else highlightedFileExt HighlightAll x
```

To avoid the behaviour of `highlightOnlyCode` and `highlightedFileExt` getting out of sync, I suggest
something like the following:

```haskell
highlightedFileExt :: HtmlHighlight ->   FileType -> String
highlightedFileExt hh  _  | not (highlightOnlyCode hh)  = "html"
highlightedFileExt _ AgdaFileType   = "html"
highlightedFileExt _ MdFileType     = "md"
highlightedFileExt _ RstFileType    = "rst"
highlightedFileExt _ TexFileType    = "tex"
```
(there might be a nicer way, this is just the idea)

Reply:
Good idea!